spec/features/ingredients/index_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe "Ingredients Index Page" do
+
+  before(:each) do
+    @milk = Ingredient.create!(name: "Milk", cost: 4)
+    @cereal = Ingredient.create!(name: "Cereal", cost: 6)
+    @banana = Ingredient.create!(name: "Banana", cost: 2)
+    @pineapple = Ingredient.create!(name: "Pineapple", cost: 4)
+    @spinach = Ingredient.create!(name: "Spinach", cost: 5)
+    @white_grape_juice = Ingredient.create!(name: "White Grape Juice", cost: 4)
+  end
+
+  it "lists all ingredients (name and cost)" do
+    visit "/ingredients"
+
+    expect(page).to have_content("Milk")
+    expect(page).to have_content(4)
+    expect(page).to have_content("Cereal")
+    expect(page).to have_content(6)
+    expect(page).to have_content("Banana")
+    expect(page).to have_content(2)
+    expect(page).to have_content("Pineapple")
+    expect(page).to have_content(4)
+    expect(page).to have_content("Spinach")
+    expect(page).to have_content(5)
+    expect(page).to have_content("White Grape Juice")
+    expect(page).to have_content(4)
+  end
```

Comment:
Try your best to avoid these ambiguous tests that could potentially give you false positives. When we're saying that we expect 4 to show up on the page, it's ambiguous because it just saying that 4 should show up on the page, it could be the temperature, the date, a phone number, etc.

I would add some string interpolation, to make sure it's clear what 4 is for, and that it's showing up where I want it to, and that I'm targeting the correct thing: `expect(page).to have_content("Cost: 4")`

If you look at your assertions, you have the same assertion for 4 listed several times. I would probably go further and use `within` blocks as well.